src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -81,26 +112,103 @@ export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
 
   const onEventBoxClick = useCallback(
     (e) => {
-      if (e.target.closest(`.${ScheduleEventBookmarkClass}`)) return;
+      if (
+        e.target.closest(`.${ScheduleEventBookmarkClass}`) ||
+        e.target.closest(`.${ScheduleEventExpandClass}`)
+      )
+        return;
 
       showEventModal();
     },
     [showEventModal]
   );
 
+  const handleMouseOver = () => {
+    if (isLongEvent) return;
+
+    toggleExpanded(true);
+  };
+
+  const handleMouseOut = () => {
+    if (isExpandedStatic) return;
+
+    toggleExpanded(false);
+  };
+
+  const handleCollapse = useCallback(() => {
+    toggleExpandedStatic(false);
+    toggleExpanded(false);
+  }, []);
+
+  const handleToggleExpand = useCallback(() => {
+    if (!isExpandedStatic) {
+      toggleExpandedStatic(!isExpandedStatic);
+
+      return;
+    }
+
+    handleCollapse();
+  }, [isExpandedStatic, handleCollapse]);
+
+  useEffect(() => {
+    function handleClickOutside(event: Event) {
+      const eventTarget = event.target as Node;
+      if (
+        eventRef.current &&
+        !eventRef?.current?.contains(eventTarget) &&
+        !isEventModalVisible
+      ) {
+        handleCollapse();
+      }
+    }
+
+    window.addEventListener("mousedown", handleClickOutside, false);
+
+    return () => {
+      window.removeEventListener("mousedown", handleClickOutside, false);
+    };
+  }, [eventRef, handleCollapse, isEventModalVisible]);
+
   return (
     <>
-      <div className={containerClasses} onClick={onEventBoxClick}>
-        <div className="ScheduleEvent__info">
-          <div className="ScheduleEvent__title">{event.name}</div>
-          <div className="ScheduleEvent__host">by {event.host}</div>
-        </div>
-
-        <div className={ScheduleEventBookmarkClass} onClick={bookmarkEvent}>
-          <FontAwesomeIcon
-            icon={event.isSaved ? solidBookmark : regularBookmark}
-          />
-        </div>
+      <div
+        ref={eventRef}
+        className={containerClasses}
+        onClick={onEventBoxClick}
+        onMouseEnter={handleMouseOver}
+        onMouseLeave={handleMouseOut}
+      >
+        {isShowExpand && (
+          <div
+            className={ScheduleEventExpandClass}
+            onClick={handleToggleExpand}
+          >
+            <FontAwesomeIcon
+              icon={squareIcon}
+              className="ScheduleEvent__expand--square"
+              style={{ width: "24px" }}
```

Comment:
Remove inline styles and use `classNames`. This applies for the rest of the components in here.

Reply:
done fd981d9

Reply:
not sure about classnames tho, it doesn't seem that it would add any value here since these are different components that need those classes, not a single one